### PR TITLE
Feature/337 allow assignments to be deleted

### PR DESCRIPTION
**Pull request by timohuber**
Depends on https://github.com/uzh/pool/pull/109

### CODE REVIEW

File: pool/web/handler/admin_experiments_assignments.ml [hunk 1]
```diff
@@ -22,65 +22,142 @@ let index req =
...
-    , HttpUtils.find_id Assignment.Id.of_string Assignment req )
-  in
-  let redirect_path =
-    Format.asprintf
-      "/admin/experiments/%s/assignments"
-      (Experiment.Id.value experiment_id)
+  let%lwt _, assignment_id, redirect_path = ids_and_redirect_from_req req in
+  let result { Pool_context.database_label; _ } =
+    Utils.Lwt_result.map_error (fun err -> err, redirect_path)
+    @@ let* assignment = Assignment.find database_label assignment_id in
```
Comment (mabiede) [inline pool/web/handler/admin_experiments_assignments.ml:101]: the only change here is the source of the variable "assignment_id" and "redirect_path". Could you change back the rest, which has just another indentation?

File: pool/web/handler/admin_experiments_assignments.ml [hunk 2]
```diff
@@ -22,65 +22,142 @@ let index req =
...
-        redirect_path
-        [ Message.set
-            ~success:[ Pool_common.Message.(Canceled Field.Assignment) ]
-        ]
-    in
-    events |>> handle
+    @@ let* assignments =
+         Assignment.find_with_follow_ups database_label assignment_id
+       in
+       let tags = Pool_context.Logger.Tags.req req in
```
Comment (mabiede) [inline pool/web/handler/admin_experiments_assignments.ml:135]: Same here, with the "logging" issue I moved the "tags" always to the top or right after the `@@` operator. Can we stick to one solution and be consistent?

File: pool/web/view/page/page_admin_assignments.ml [hunk 3]
```diff
@@ -26,28 +57,68 @@ module Partials = struct
...
     session
     assignments
     =
-    let cancelable = Session.assignments_cancelable session |> CCResult.is_ok in
-    let action assignment =
-      Format.asprintf
-        "/admin/experiments/%s/assignments/%s/cancel"
-        (experiment_id |> Experiment.Id.value)
-        (assignment.Assignment.id |> Pool_common.Id.value)
+    let deletable m = m |> Assignment.is_deletable |> CCResult.is_ok in
```
Comment (mabiede) [inline pool/web/view/page/page_admin_assignments.ml:66]:
```suggestion
    let deletable = CCFun.(Assignment.is_deletable %> CCResult.is_ok) in
```

File: pool/web/view/page/page_admin_assignments.ml [hunk 4]
```diff
@@ -26,28 +57,68 @@ module Partials = struct
...
+      let open Pool_common in
+      button_form
+        "cancel"
+        I18n.CancelAssignment
+        (Message.Cancel None)
+        `CloseCircle
+    in
+    let mark_as_deleted =
+      let open Pool_common in
+      (* TODO[timhub]: Add hint, only if this assignment has a follow-up
```
Comment (mabiede) [inline pool/web/view/page/page_admin_assignments.ml:107]: todo

**Review by mabiede — CHANGES_REQUESTED** [review]

File: pool/web/view/page/page_admin_assignments.ml (hunk 4 above)
Comment (timohuber) [reply pool/web/view/page/page_admin_assignments.ml]: What do you think of this solution? https://github.com/uzh/pool/pull/110/commits/0ba62c704de30bb4f3e30374ba22bbab636eecba

Comment (timohuber) [reply pool/web/view/page/page_admin_assignments.ml]: @mabiede ping

**Review by mabiede — APPROVED** [review]

File: pool/web/view/page/page_admin_assignments.ml (hunk 4 above)
Comment (mabiede) [reply pool/web/view/page/page_admin_assignments.ml]: Nice, a small update that makes things easier.

**Pull request merged**